index.js

```diff
@@ -314,11 +313,7 @@ RedisClient.prototype.init_parser = function () {
 
     // "reply error" is an error sent back by Redis
     this.reply_parser.on("reply error", function (reply) {
-        if (reply instanceof Error) {
-            self.return_error(reply);
-        } else {
-            self.return_error(new Error(reply));
-        }
+        self.return_error(reply);
```

Comment:
hiredis did not have any non-Error and as it was the only spot in the js parser that was not an Error it should be fixed properly. And all tests still pass (they would not have without fixing the non-Error).